jabgui/src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -133,4 +151,726 @@ void findAllAssociatedNotLinkedFilesInsteadOfTheFirstOne(@TempDir Path tempDir)
                 new LinkedFile("", newPath2String, "PDF"));
         assertEquals(expected, Set.copyOf(matchedFiles));
     }
+
+    @Test
+    void findAllAssociatedNotLinkedFilesAndNotRepeated(@TempDir Path tempDir) throws IOException {
+        when(autoLinkPrefs.getCitationKeyDependency()).thenReturn(AutoLinkPreferences.CitationKeyDependency.START);
+
+        // File and folder
+        Path subdir = tempDir.resolve("subdir");
+        Files.createDirectory(subdir);
+        Path fileA = tempDir.resolve("CK_A.pdf");
+        Files.createFile(fileA);
+        Path fileB = tempDir.resolve("CK_B.pdf");
+        Files.createFile(fileB);
+        BibEntry entry = new BibEntry(StandardEntryType.Misc)
+                .withFiles(List.of(
+                        new LinkedFile("", "subdir/CK_B.pdf", "PDF")
+                ));
+        entry.setCitationKey("CK");
+
+        BibDatabaseContext context = mock(BibDatabaseContext.class);
+        when(context.getFileDirectories(filePreferences)).thenReturn(List.of(tempDir));
+        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
+                context,
+                externalApplicationsPreferences,
+                filePreferences,
+                autoLinkPrefs);
+
+        // find by citation will return CK_A.pdf and CK_B.pdf
+        // find by broken linked file name will return CK_B.pdf
+        Collection<LinkedFile> matchedFiles = util.findAssociatedNotLinkedFiles(entry);
+        Set<LinkedFile> expected = Set.of(
+                new LinkedFile("", "CK_A.pdf", "PDF"),
+                new LinkedFile("", "CK_B.pdf", "PDF"));
+        assertEquals(expected, Set.copyOf(matchedFiles));
+    }
+
+    //******************************************************************************************************
+    //*********************************** Test linkAssociatedFiles *****************************************
+    //******************************************************************************************************
```

Comment:
The following test cases are grouped by commets like this

Reply:
You can use nested tests to structure your tests more easily.
https://docs.junit.org/6.0.1/writing-tests/nested-tests.html

Reply:
Thanks! I’ve already applied `@Nested`; with code folding, the test structure is clean and clear.